crates/data_components/src/oracle/execution_plan.rs

```diff
@@ -0,0 +1,211 @@
+/*
+Copyright 2024-2025 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use std::{any::Any, fmt, sync::Arc};
+
+use arrow::datatypes::SchemaRef;
+use datafusion::{
+    error::{DataFusionError, Result as DataFusionResult},
+    execution::TaskContext,
+    logical_expr::Expr,
+    physical_expr::EquivalenceProperties,
+    physical_plan::{
+        DisplayAs, DisplayFormatType, ExecutionPlan, Partitioning, PlanProperties,
+        SendableRecordBatchStream,
+        execution_plan::{Boundedness, EmissionType},
+        stream::RecordBatchStreamAdapter,
+    },
+    sql::TableReference,
+};
+
+pub type Result<T, E = super::Error> = std::result::Result<T, E>;
+
+use oracle::Row;
+use snafu::ResultExt;
+
+use crate::oracle::{QuerySnafu, connection::OracleConnectionPool, convert::rows_to_arrow};
+use futures::{StreamExt, TryStreamExt};
+
+#[derive(Clone)]
+pub struct OracleExecPlan {
+    projected_schema: SchemaRef,
+    table_reference: TableReference,
+    pool: Arc<OracleConnectionPool>,
+    _filters: Vec<Expr>,
+    _limit: Option<usize>,
+    properties: PlanProperties,
+}
+
+impl OracleExecPlan {
+    pub fn new(
+        projections: Option<&Vec<usize>>,
+        schema: &SchemaRef,
+        table_reference: &TableReference,
+        pool: Arc<OracleConnectionPool>,
+        filters: &[Expr],
+        limit: Option<usize>,
+    ) -> DataFusionResult<Self> {
+        let projected_schema = project_schema_safe(schema, projections)?;
+
+        Ok(Self {
+            projected_schema: Arc::clone(&projected_schema),
+            table_reference: table_reference.clone(),
+            pool,
+            _filters: filters.to_vec(),
+            _limit: limit,
+            properties: PlanProperties::new(
+                EquivalenceProperties::new(projected_schema),
+                Partitioning::UnknownPartitioning(1),
+                EmissionType::Incremental,
+                Boundedness::Bounded,
+            ),
+        })
+    }
+
+    #[allow(clippy::unnecessary_wraps)]
+    pub fn sql(&self) -> DataFusionResult<String> {
```

Comment:
To be extended with filter expressions (where clause) and limit support next, including tests